### DISCUSSION TITLE
XYChart range

### DISCUSSION THREAD

**Discussion by jirikolarik** [airbnb/visx#1147] [Q&A]
Hey,

I'm trying to build a chart with XYChart where I can set a visible range. Let say I have AreaSeries with a date on x axis and values ranging from 200 to 300 on y axis, but I need to display the whole chart with a range from 0 to 400.

**Comment by williaster** [comment]
Hey @jirikolarik 👋   thanks for checking out `visx`. Scales in `XYChart` are controlled by the `xScale/yScale` config objects that you pass in. These configs match those in `@visx/scale` exactly, so you can check out [those docs](https://airbnb.io/visx/docs/scale) for more info if you're not familiar (these basically are `d3-scale` scales, but accept and object configs rather than the less-intuitive chaining/jQuery-like config that d3 uses).

To specify a custom data range (this is called a `domain` in d3 nomenclature), you'd simply do something like `yScale={{ type: "linear", domain: [-5000, 5000] }}`. Here's a [codesandbox demo](https://codesandbox.io/s/custom-chart-scale-domain-w2p9q?file=/Example.tsx:1676-1726).

<img src="https://user-images.githubusercontent.com/4496521/114069788-38a5d900-9854-11eb-81a5-a254e2d8eb8a.png" width="500" />

**Reply by jirikolarik** [reply]
Hey @williaster and thank you for your exhausting answer!:)

In this case it really works, but when I set the domain to [2000, 5000] `yScale={{ type: "linear", domain: [2000, 5000] }}` then the `yScale` doesn't start on `2000` but rather on `0`. 

![Screenshot from 2021-04-15 08-10-51](https://user-images.githubusercontent.com/3116918/114822664-b3c23e80-9dc2-11eb-99b1-75bbdc6bf202.png)

**Reply by williaster** [reply]
ah, that's because linear scales have `zero: true` by default (because it's often best practice to include zero in the y range so it's not misleading, tho there are clearly exceptions). so you'd just need to add `{ zero: false }`. In the sandbox example I would also add `{ clamp: true }` because there are data points `< 2k` which render outside the chart otherwise.

**Reply by jirikolarik** [reply]
Oh wow, thanks, you're an absolute life-saver! I wouldn't figured that out by myself :)

**Reply by jirikolarik** [reply]
Hey @williaster, I hope you don't mind getting back on this topic. In this case the char always begins at 0, how can I make it to begin on the lowest point of the y axis?

Let's say I turned BarSeries into AreaSeries (https://codesandbox.io/s/custom-chart-scale-domain-forked-8slxu) and I want to fill the area black, but not from 0, but all the way from the bottom?

**Reply by williaster** [reply]
Hey @jirikolarik 👋  most of the series go from `0` to the y-value of `Datum` because starting below zero could be misleading, can you elaborate on the use case / why you want this effect?

In any case, `AreaSeries` have an optional [`y0Accessor`](https://airbnb.io/visx/docs/xychart#AreaSeries_y0Accessor) (and `x0Accessor` if the orientation is flipped) which specifies the baseline of the area shape (it defaults to `0`). 

Here's a [forked sandbox](https://codesandbox.io/s/custom-chart-scale-domain-forked-x39x9?file=/Example.tsx) which just specifies `y0Accessor={() => -5000}`

<img src="https://user-images.githubusercontent.com/4496521/129785251-856ca613-24d0-4276-9e49-85ef66500e3d.png" width="600" />

**Reply by suvrabha** [reply]
Getting error  Type '"linear"' is not assignable to type '"band"'when I specify domain in yScale

**Reply by KuNman** [reply]
Hey @williaster , thanks for example, but could you demo how can I set starting point for y axis as for example 1000? I get empty space between 0-1000 value as right axis doesn't autoscale to minimum value, it's always hardcoded to 0.
![Screenshot 2022-10-12 at 02 46 19](https://user-images.githubusercontent.com/25631854/195223733-2a55d0fb-afbe-4929-bbb9-21294dc390eb.png)

**Reply by williaster** [reply]
I believe this is likely due to `zero: true` being the default for linear scales. Here's an update where I set the domain to `[1000, 5000]` and `zero: false` https://codesandbox.io/s/custom-chart-scale-domain-forked-x39x9?file=/Example.tsx

<img width="665" alt="image" src="https://user-images.githubusercontent.com/4496521/196790139-f12c8f16-d7a6-480a-a87d-580b319881fe.png">
